aii-ks/src/main/perl/ks.pm

```diff
@@ -1132,7 +1139,11 @@ success() {
 
     subject="[\\`date +'%x %R %z'\\`] Quattor installation on $fqdn succeeded"
     if [ -x /usr/bin/mailx ]; then
-        env MAILRC=/dev/null from=root\@$fqdn $mailx_smtp mailx -s "\\\$subject" $rootmail <<End_of_mailx
+        mailx_options="from=root\@$fqdn $mailx_smtp mailx"
```

Comment:
There's now considerable duplication here and no real difference between this and lines 1105-1109. Perhaps it's time to factor this out into a function called send_email(subject)?

Reply:
I understand what you mean it's 3 extra line of duplication in fail and success function.
And it's not just the subject, body is different from success and failure. given the benefit I'm not sure its worth the effort.

Reply:
It's not so much the extra lines but the sheer complexity of these lines now -- there are benefits of abstracting that out to make the code easier to follow too and make is easier to change again in the future. You could also encapsulate all the mail config into one place, right now it's defined in multiple places (e.g. 1080-1084).

Having said that, this sort of low level embedded scripting has its own complexity of needing to be generated on the fly from within another programme--getting the escaping right is a nightmare so I appreciate if you don't want to add further levels of indirection.

I'll defer to @jrha  and @stdweird as to which they prefer.

Reply:
Definitely pros and cons either way, but I don't think it's worth further refactoring at this point.